Tella/Data/Database/TellaDataBase.swift

```diff
@@ -509,24 +509,36 @@ class TellaDataBase {
         
         guard let reportId, let report = self.getReport(reportId: reportId) else { return 0}
         
-        deleteReportFiles(report: report)
-        
+        deleteReportFiles(reportIds: [reportId])
+
         let reportCondition = [KeyValue(key: D.cReportId, value: report.id as Any)]
         return try statementBuilder.delete(tableName: D.tReport,
                                            primarykeyValue: reportCondition)
     }
-    
-    func deleteReportFiles(report:Report) {
+
+    func deleteSubmittedReport() throws -> Int {
+
+        let submittedReports = self.getReports(reportStatus: [.submitted])
+        let reportIds = submittedReports.compactMap{$0.id}
+
+        deleteReportFiles(reportIds: reportIds)
+
+        let reportCondition = [KeyValue(key: D.cStatus, value: ReportStatus.submitted.rawValue)]
+        
+        return try statementBuilder.delete(tableName: D.tReport,
+                                           primarykeyValue: reportCondition)
+    }
+
```

Comment:
Maybe we should change the name to `deleteSubmittedReportsCount` as the function is also returning the number of deleted reports.